lib/improver/utilities/cube_manipulation.py

```diff
@@ -289,30 +301,40 @@ def _equalise_cube_attributes(cubes):
     # without raising a warning message
     silent_attributes = ['history', 'title', 'mosg__grid_version']
     unmatching_attributes = compare_attributes(cubes)
+
     if len(unmatching_attributes) > 0:
         for i, cube in enumerate(cubes):
             # Remove ignored attributes.
             for attr in silent_attributes:
                 if attr in unmatching_attributes[i]:
                     cube.attributes.pop(attr)
                     unmatching_attributes[i].pop(attr)
-            # Add associated model_id if model_configurations do not match.
-            if "mosg__model_configuration" in unmatching_attributes[i]:
-                model_title = cube.attributes.pop('mosg__model_configuration')
-                for key, val in MODEL_ID_DICT.items():
-                    if val == model_title:
-                        break
-                new_model_id_coord = build_coordinate([key],
+
+            # If a model_id_attr has been specified, assume we are trying to
+            # grid blend and throw an error if the model_id_attr does not
+            # match that on the cubes to be blended.
+            if model_id_attr is not None and \
+                    model_id_attr not in cube.attributes:
+                msg = ('Cannot create model ID coordinate for grid blending '
+                       'as the model ID attribute specified does not match '
+                       'that on the cubes to be blended')
+                raise ValueError(msg)
+
+            if model_id_attr in unmatching_attributes[i]:
+                model_title = cube.attributes.pop(model_id_attr)
```

Comment:
Line 173 in weighted_blend.py changes the attribute "mosg__model_configuration" to be "blend" in cases where we blend over the model coordinate. This can be seen in the existing CLI test KGO for the model blending case (```weighted_blending/model/kgo.nc```).

The change here has means the following:
1. If we run IMPROVER with the model_id_attr set, as it is by default in the CLI changes you have made, we will remove the mosg__model_configuration attribute at this point and then add it back in in weighted_blend assuming there are other attributes starting with the prefix mosg_ and we blend over a 'model_id' type coordinate.
2, If we run the code with non-MO data, the attribute will be deleted here and nothing will be added back in weighted_blend as unfortunately the "mosg__model_configuration" attribute key is hardcoded.

An alternative might be to change the attribute at this stage to become 'blend' (rather than remove it) if we can only foresee the conditions here being fulfilled as the result of us grid blending. We could then make this change for the "model_id_attr" attribute whether it is Met Office or not. We could also remove the horrible bit of hardcoding in weighted_blend.

This would then mean that the output of your new CLI test would include the additional attribute ```non_mo_grid_domain: blend```, which would better match the other test for model blending.

I think this needs some discussion with others.

Reply:
Changes made as per discussion and confirmed in unit tests that model ID attribute gets set to 'blend' for MO and non MO model blending.

Reply:
I cannot see this change Mark. I'm not sure if it was not pushed up?

Reply:
I've not made the relevant changes, including additional unit tests in a few places.